src/lib/state/actions/query.js

```diff
@@ -137,37 +148,104 @@ export const resetQuery = () => {
   };
 };
 
+/**
+ * Update URL parameters.
+ * @param {object} queryState - current query state
+ * @param {object} appConfig - app config
+ * @param {boolean} shouldReplaceUrlQueryString - true if should replace the last browser history state
+ * @param {boolean} shouldUpdateUrlQueryString - true if it should add a new browser history state
+ */
+const updateURLParameters = (
+  queryState,
+  appConfig,
+  shouldReplaceUrlQueryString,
+  shouldUpdateUrlQueryString
+) => {
+  const urlHandlerApi = appConfig.urlHandlerApi;
+  if (urlHandlerApi) {
+    if (shouldReplaceUrlQueryString) {
+      urlHandlerApi.replace(queryState);
+    } else if (shouldUpdateUrlQueryString) {
+      urlHandlerApi.set(queryState);
+    }
+  }
+};
+
+/**
+ * Update query state and URL args with the new query state given by the backend response.
+ * @param {object} response - API response
+ * @param {func} dispatch - Redux `dispath` function
+ * @param {func} getState - function to get the Redux state
+ * @param {object} appConfig - app config
+ */
+const updateQueryStateAfterResponse = (
+  response,
+  dispatch,
+  getState,
+  appConfig
+) => {
+  dispatch({
+    type: SET_QUERY_STATE,
+    payload: response.newQueryState,
+  });
+
+  const urlHandlerApi = appConfig.urlHandlerApi;
+  if (urlHandlerApi) {
+    // Replace the URL args with the response new query state
+    const updatedQueryState = _cloneDeep(getState().query);
+    urlHandlerApi.replace(updatedQueryState);
+  }
+  delete response.newStateQuery;
+};
+
+const updateQueryStateSorting = (queryState, appState, appConfig) => {
+  if (_isEmpty(appConfig.defaultSortingOnEmptyQueryString)) {
+    return;
+  }
+
+  const userHasChangedSorting = appState.hasUserChangedSorting;
+  if (userHasChangedSorting === false) {
+    const isQueryStringEmpty = queryState.queryString === '';
+    if (isQueryStringEmpty) {
+      queryState.sortBy = appConfig.defaultSortingOnEmptyQueryString.sortBy;
+      queryState.sortOrder =
+        appConfig.defaultSortingOnEmptyQueryString.sortOrder;
+    } else {
+      queryState.sortBy = appState.initialSortBy;
+      queryState.sortOrder = appState.initialSortOrder;
+    }
+  }
+};
+
+/**
+ * Execute search API request with current query state and update results state with response.
+ * @param {object} options - `shouldUpdateUrlQueryString` and `shouldReplaceUrlQueryString` to choose if the
+ *                           browser history state should be updated with the current query state.
+ */
 export const executeQuery = ({
   shouldUpdateUrlQueryString = true,
   shouldReplaceUrlQueryString = false,
 } = {}) => {
   return async (dispatch, getState, config) => {
-    let queryState = _cloneDeep(getState().query);
+    const appState = getState().app;
+    let queryState = getState().query;
+    updateQueryStateSorting(queryState, appState, config);
+
+    queryState = _cloneDeep(queryState);
```

Comment:
Maybe better to return a cloned state from the `updateQueryStateSorting `? Then, if someone uses it in another action, there wouldn't need to care about the mutation that is happening inside the function...

Reply:
I have changed, can you double check? I am not sure I fully understood your comment.